### PR TITLE
refact: Use Background Publisher instead of Push To Core

### LINKED ISSUE #56: [LLRP] Use Background Publisher to send Inventory Events
Currently the inventory events are sent via the Push To Core functionality. The newer `Background Publisher` is a better option now that Core Data is an optional service.

### PR DESCRIPTION
**Pull request by lenny-intel**
closes #56

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-rfid-llrp-inventory/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **Requires integration testing**
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) **No doc updates required**

## Testing Instructions
**TBD**

## New Dependency Instructions (If applicable)
**N/A**

### CODE REVIEW

**Review by marcpfuller — CHANGES_REQUESTED** [review]

File: internal/inventory/app/controller.go [hunk 1]
```diff
@@ -294,29 +297,35 @@ func (app *InventoryApp) persistSnapshot(snapshot []inventory.StaticTag) {
...
+		app.lc.Debugf("Publishing Inventory Event of type %s: %+v", resourceName, event)
 		edgeXEvent.AddObjectReading(resourceName, event)
 	}
 
-	ctx, cancel := context.WithTimeout(ctx, coreDataPostTimeout)
-	defer cancel()
-
-	if _, err := app.service.EventClient().Add(ctx, requests.NewAddEventRequest(edgeXEvent)); err != nil {
-		errs = append(errs, errors.Wrap(err, "unable to push inventory event(s) to core-data"))
+	// TODO : wrap in AddEvent request.
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply internal/inventory/app/controller.go]: Nope. Removed

**Review by marcpfuller — DISMISSED** [review]
LGTM

File: internal/inventory/app/app.go [hunk 2]
```diff
@@ -90,6 +91,11 @@ func (app *InventoryApp) Initialize() error {
 		return errors.Wrap(err, "failed to listen for custom config changes")
 	}
 
+	app.publisher, err = app.service.AddBackgroundPublisher(1)
+	if err != nil {
+		return errors.Wrap(err, "failed to add BackGroundPublisher")
```
Comment (ajcasagrande) [inline internal/inventory/app/app.go:96]:
```suggestion
		return errors.Wrap(err, "failed to add BackgroundPublisher")
```

**Review by ajcasagrande — DISMISSED** [review]
LGTM, just one typo.

File: internal/inventory/app/app.go (hunk 2 above)
Comment (lenny-intel) [reply internal/inventory/app/app.go]: fixed

**Review by ajcasagrande — APPROVED** [review]

**Pull request merged**